src/pytorch_lightning/demos/mnist_datamodule.py

```diff
@@ -201,7 +199,7 @@ def setup(self, stage: Optional[str] = None) -> None:
         """Split the train and valid dataset."""
         extra = dict(transform=self.default_transforms) if self.default_transforms else {}
         dataset: Dataset = MNIST(self.data_dir, train=True, download=False, **extra)
-        train_length = len(dataset)
+        train_length = len(dataset.data)
```

Comment:
```suggestion
        train_length = len(dataset)
```
I don't think this is required.